src/Scouting.elm

```diff
@@ -1,65 +1,121 @@
-module Scouting exposing (Msg, Model, init, update, main, view )
+module Scouting exposing (Model, Msg, init, main, update, view)
 
 import Browser
-import Html.Attributes exposing (placeholder, value)
-import Html exposing (input , text)
-import Html.Events exposing (onClick , onInput)
+import Html exposing (input, text)
+import Html.Attributes exposing (placeholder, style, value)
+import Html.Events exposing (onClick, onInput)
+import Maybe.Extra exposing (unwrap)
 
 
-inputwriter : String -> (String -> Msg) -> String -> Html.Html Msg
-inputwriter placeholderInput msg model =
-    Html.div [] [
-    input [placeholder placeholderInput, onInput msg, value model] []
-    ,text "\n"  
-    ]
+inputWriter : String -> (String -> Msg) -> String -> Html.Html Msg
+inputWriter placeholderInput msg model =
+    Html.div [ style "margin-bottom" "1%" ]
+        [ input [ placeholder placeholderInput, onInput msg, value model ]
+            []
+        ]
+
+
+unwrapToString : Maybe Int -> String
+unwrapToString maybeInt =
+    unwrap "" String.fromInt maybeInt
+
+
+type Pressed
+    = Pressed
+    | NotPressed
+    | InputError
+
 
-type Msg =
-    ScoutersNameInput String
+inputChecker : Model -> Model
+inputChecker model =
+    if model.scoutersName == "" || model.scoutedTeamNumber == Nothing || model.scoutedTeamDriverStationPosition == "" || model.matchNumber == Nothing then
+        { model | isPressed = InputError }
+
+    else
+        { model | isPressed = Pressed }
+
+
+type Msg
+    = ScoutersNameInput String
     | TeamNumberInput String
     | MatchNumberInput String
     | DriverStationInput String
-    | Page Bool
+    | Page Pressed
 
 
 type alias Model =
-    {scoutersName : String
-    , scoutedTeamNumber : String
-    , matchNumber : String
+    { scoutersName : String
+    , scoutedTeamNumber : Maybe Int
+    , matchNumber : Maybe Int
     , scoutedTeamDriverStationPosition : String
```

Comment:
Should this really be a `String`?
Are `""`, `"abcd"`, `"I ♥ Elm"` all valid `scoutedTeamDriverStationPosition`s?
Also, you can shorten the name to just `driverStationPosition` - it's just as clear and much more convenient.

Reply:
The bit about naming applies to all fields (i.e. a `scouter` that's a `String` is clearly the scouter's name).